gubernator/github/classifier.py

```diff
@@ -71,37 +71,28 @@ def classify_issue(repo, number):
         .fetch(keys_only=True))
 
     logging.debug('classifying %s %s (%d events)', repo, number, len(event_keys))
-    event_tuples = []
-    last_event_timestamp = datetime.datetime(2000, 1, 1)
+    last_event_timestamp = [datetime.datetime(2000, 1, 1)]
 
+    def events_iterator():
+        for x in xrange(0, len(event_keys), 100):
+            events = ndb.get_multi(event_keys[x:x+100])
+            for event in events:
+                last_event_timestamp[0] = max(last_event_timestamp[0], event.timestamp)
+            yield [event.to_tuple() for event in events]
 
-    if len(event_keys) > 800:
-        logging.warning('too many events. blackholing.')
-        return False, False, [], {'num_events': len(event_keys)}, last_event_timestamp
-
-    deduper = Deduper()
-
-    for x in xrange(0, len(event_keys), 100):
-        events = ndb.get_multi(event_keys[x:x+100])
-        last_event_timestamp = max(last_event_timestamp, max(e.timestamp for e in events))
-        event_tuples.extend([deduper.dedup(event.to_tuple()) for event in events])
-
-    del deduper  # attempt to save memory
-    del events
-
-    merged = get_merged(event_tuples)
-    statuses = None
-    if 'head' in merged:
+    def get_status_for(sha):
         statuses = {}
-        for status in models.GHStatus.query_for_sha(repo, merged['head']['sha']):
-            last_event_timestamp = max(last_event_timestamp, status.updated_at)
+        for status in models.GHStatus.query_for_sha(repo, sha):
+            last_event_timestamp[0] = max(last_event_timestamp[0], status.updated_at)
             statuses[status.context] = [
                 status.state, status.target_url, status.description]
+        return statuses
 
-    return list(classify(event_tuples, statuses)) + [last_event_timestamp]
+    classified = classify(None, events_iterator(), status_fetcher=get_status_for)
+    return list(classified) + last_event_timestamp
 
 
-def get_merged(events):
+def get_merged(events, merged=None):
     '''
```

Comment:
Please follow style guide for pydoc:
```
def whatever():
   """One line summary.

  extra stuff.
  """
```

https://google.github.io/styleguide/pyguide.html?showone=Comments#Comments

Reply:
Seems unaddressed

Reply:
Fixed.